src/printer.rs

```diff
@@ -289,21 +292,11 @@ impl<'a> Printer for InteractivePrinter<'a> {
 
         if self.config.style_components.grid() {
             self.print_horizontal_line(handle, '┬')?;
-
-            write!(
```

Comment:
I'm a bit short on time, so please accept my apology in advance in asking this question rather than figuring it out for myself:

Can we somehow refactor the code so that we don't need to check for `if self.config.style_components.grid()` twice? That would simplify the code which is always nice.

Eventually I will get time to figure it out for myself, but that can take a while.

Reply:
I tried to do that initially, but didn't find a straightforward way to do that. I can put comments on what each one of those if blocks does and why they exist separately. Basically the first one is only required before the first grid line to insert the cornering grid, or else insert a new-line break between files.

I moved the outside check closer to the loop as well which I think makes reasoning about them easier.